src/v/cluster/members_manager.cc

```diff
@@ -525,6 +569,71 @@ members_manager::handle_configuration_update_request(
           return ss::make_ready_future<ret_t>(
             errc::join_request_dispatch_error);
       });
-} // namespace cluster
+}
+
+ss::future<result<decommissioning_status>>
+members_manager::decommission_nodes(std::vector<model::node_id> ids) {
+    using ret_t = result<decommissioning_status>;
+    std::vector<decommissioning_status::node_state> res;
+    res.reserve(ids.size());
+
+    vlog(clusterlog.trace, "decomissioning nodes {}", ids);
+    if (!_raft0->is_leader()) {
```

Comment:
move earlier, so the reserve call isn't performed if your going to exit early anyway